js/src/wrappers/anthropic.test.ts

```diff
@@ -232,4 +243,254 @@ describe("anthropic client unit tests", () => {
     expect(metrics.start < metrics.end).toBe(true);
     expect(metrics.end <= endTime).toBe(true);
   });
+
+  test("test client.beta.messages.create", async () => {
+    let startTime: number = -1;
+    let endTime: number = -1;
+
+    // wrapped has to be last so the timing test works
+    const clients = [anthropic, client];
+    const responses = await Promise.all(
+      clients.map(async (c) => {
+        startTime = getCurrentUnixTimestamp();
+
+        const response = await c.beta.messages.create({
+          model: TEST_MODEL,
+          messages: [{ role: "user", content: "What's 4*4?" }],
+          max_tokens: 100,
+          system: "Return the result only.",
+          temperature: 0.1,
+        });
+        endTime = getCurrentUnixTimestamp();
+        return response;
+      }),
+    );
+
+    // validate that the wrapped and unwrapped clients produce the same response
+    assertAnthropicResponsesEqual(responses);
+
+    // validate we traced the second request
+    const spans = await backgroundLogger.drain();
+    expect(spans).toHaveLength(1);
+    const span = spans[0] as any;
```

Comment:
I believe we could have many of these. You could try something like:

https://github.com/braintrustdata/braintrust-sdk/blob/89a733ea8deaa0ac4a1bfb99c65e9603ff55f735/integrations/langchain-js/src/test/utils.ts#L35

to convert individual logs to spans (essentially a merge) to then do a proper check against the shape of the spans (i.e. root, child, subchild and so on)

Reply:
see example: https://github.com/braintrustdata/braintrust-sdk/blob/89a733ea8deaa0ac4a1bfb99c65e9603ff55f735/integrations/langchain-js/src/BraintrustCallbackHandler.test.ts#L66

Reply:
Ok, I think this is a good idea, but I'm going to leave it for a future PR because I think this test covers a lot of it. Next time I'll tackle this.